server/src-lib/Hasura/Eventing/ScheduledTrigger.hs

```diff
@@ -0,0 +1,435 @@
+          Right _ -> pure ()
+          Left err ->
+            L.unLogger logger $ ScheduledTriggerInternalErr $ err500 Unexpected (T.pack $ show err)
+    sleep (minutes 1)
+    where
+      getDeprivedScheduledTriggerStats = liftTx $ do
+        map uncurryStats <$>
+          Q.listQE defaultTxErrorHandler
+          [Q.sql|
+           SELECT name, upcoming_events_count, max_scheduled_time
+            FROM hdb_catalog.hdb_scheduled_events_stats
+            WHERE upcoming_events_count < 100
+           |] () True
+
+      uncurryStats (n, count, maxTs) = ScheduledTriggerStats n count maxTs
+
+      withST stis stat = do
+        case Map.lookup (stsName stat) stis of
+          Nothing -> do
+            L.unLogger logger $
+              ScheduledTriggerInternalErr $ err500 Unexpected "could not find scheduled trigger in the schema cache"
+            pure Nothing
+          Just sti -> pure $ Just (sti, stat)
+
+insertScheduledEventsFor :: [(ScheduledTriggerInfo, ScheduledTriggerStats)] -> Q.TxE QErr ()
+insertScheduledEventsFor scheduledTriggersWithStats = do
+  let scheduledEvents = flip concatMap scheduledTriggersWithStats $ \(sti, stats) ->
+        generateScheduledEventsFrom (stsMaxScheduledTime stats) sti
+  case scheduledEvents of
+    []     -> pure ()
+    events -> do
+      let insertScheduledEventsSql = TB.run $ toSQL
+            SQLInsert
+              { siTable    = scheduledEventsTable
+              , siCols     = map unsafePGCol ["name", "scheduled_time"]
+              , siValues   = ValuesExp $ map (toTupleExp . toArr) events
+              , siConflict = Just $ DoNothing Nothing
+              , siRet      = Nothing
+              }
+      Q.unitQE defaultTxErrorHandler (Q.fromText insertScheduledEventsSql) () False
+  where
+    toArr (ScheduledEventSeed n t) = [(triggerNameToTxt n), (formatTime' t)]
+    toTupleExp = TupleExp . map SELit
+
+insertScheduledEvents :: [ScheduledEventSeed] -> Q.TxE QErr ()
+insertScheduledEvents events = do
+  let insertScheduledEventsSql = TB.run $ toSQL
+        SQLInsert
+          { siTable    = scheduledEventsTable
+          , siCols     = map unsafePGCol ["name", "scheduled_time"]
+          , siValues   = ValuesExp $ map (toTupleExp . toArr) events
+          , siConflict = Just $ DoNothing Nothing
+          , siRet      = Nothing
+          }
+  Q.unitQE defaultTxErrorHandler (Q.fromText insertScheduledEventsSql) () False
+  where
+    toArr (ScheduledEventSeed n t) = [(triggerNameToTxt n), (formatTime' t)]
+    toTupleExp = TupleExp . map SELit
+
+generateScheduledEventsFrom :: UTCTime -> ScheduledTriggerInfo-> [ScheduledEventSeed]
+generateScheduledEventsFrom startTime ScheduledTriggerInfo{..} =
+  let events =
+        case stiSchedule of
+          AdHoc _   -> empty -- ad-hoc scheduled events are created through 'create_scheduled_event' API
+          Cron cron -> generateScheduleTimes startTime 100 cron -- by default, generate next 100 events
```

Comment:
The comment here says “by default,” but 100 seems to be hard-coded, so it will never be anything else. Is that wrong?

Reply:
The comment was wrong, it will always be 100. Thanks for pointing out the mistake 🙂